nemo/collections/asr/parts/utils/streaming_utils.py

```diff
@@ -1161,3 +1162,214 @@ def transcribe(
         for idx in range(self.batch_size):
             output.append(self.greedy_merge(self.unmerged[idx]))
         return output
+
+
+class FramewiseStreamingAudioBuffer:
+    """
+    A buffer to be used for frame-wise streaming. It can load a single or multiple audio files/processed signals, split them in chunks and return one on one.
+    It can be used to simulate streaming audio or audios.
+    """
+
+    def __init__(self, model, online_normalization=None):
+        '''
+        Args:
+            model: An ASR model.
+            online_normalization (bool): whether to perform online normalization per chunk or normalize the whole audio before chunking
+        '''
+        self.model = model
+        self.buffer = None
+        self.buffer_idx = 0
+        self.streams_length = None
+        self.step = 0
+
+        self.online_normalization = online_normalization
+        if not isinstance(model.encoder, StreamingEncoder):
+            raise ValueError(
+                "The model's encoder is not inherited from StreamingEncoder, and likely not to support streaming!"
+            )
+        if model.encoder.streaming_cfg is None:
+            model.encoder.setup_streaming_params()
+        self.streaming_cfg = model.encoder.streaming_cfg
+
+        self.input_features = model.encoder._feat_in
+
+        self.preprocessor = self.extract_preprocessor()
+
+    def __iter__(self):
+        while True:
+            if self.buffer_idx >= self.buffer.size(-1):
+                return
+                # raise StopIteration
+            if self.buffer_idx == 0 and isinstance(self.streaming_cfg.chunk_size, list):
+                chunk_size = self.streaming_cfg.chunk_size[0]
+            else:
+                chunk_size = (
+                    self.streaming_cfg.chunk_size[1]
+                    if isinstance(self.streaming_cfg.chunk_size, list)
+                    else self.streaming_cfg.chunk_size
+                )
+
+            if self.buffer_idx == 0 and isinstance(self.streaming_cfg.shift_size, list):
+                shift_size = self.streaming_cfg.shift_size[0]
+            else:
+                shift_size = (
+                    self.streaming_cfg.shift_size[1]
+                    if isinstance(self.streaming_cfg.shift_size, list)
+                    else self.streaming_cfg.shift_size
+                )
+
+            audio_chunk = self.buffer[:, :, self.buffer_idx : self.buffer_idx + chunk_size]
+
+            zeros_pads = None
```

Comment:
Add some high level comments as to what is going on.

Reply:
added.